[CWS] SecAgent config subcommand to fx

modernplumbing opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

Convert the SecAgent config subcommand to fx.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

modernplumbing committed:
moving root command

subcommand

modernplumbing committed:
converting app/start.go and app/app.go to start/command.go (and command.go in previous commit)

start

start test

start

start

start command var change

start cmd import

modernplumbing committed:
compliance subcmd using log and config components

modernplumbing committed:
compliance, runtime, and check commands using command instead of common

modernplumbing committed:
runtime using components

modernplumbing committed:
moving logs context to command from common, and deleting duplicated logs context code in runtime

modernplumbing committed:
fixing rebase clobbers

modernplumbing committed:
param setup

modernplumbing committed:
handle case with no config files

modernplumbing committed:
squashing more usages of pkg/util/log and pkg/config

modernplumbing committed:
moving ConfigParams in compliance check fnc until after cfgpath has been parsed

modernplumbing committed:
convert config subcommand to fx

modernplumbing committed:
adding cluster agent bool to check entrypoint

modernplumbing committed:
using log component in logs context

modernplumbing committed:
fixing no api key error message

modernplumbing committed:
adding log.Flush() to one shot funcs. TODO: remove once log component has self-flushing capabilities

modernplumbing commented:
Current dependencies on/for this PR:
* main
  * **PR #14814** <a href="https://app.graphite.dev/github/pr/DataDog/datadog-agent/14814" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
    * **PR #15153** <a href="https://app.graphite.dev/github/pr/DataDog/datadog-agent/15153" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈
    * **PR #14896** <a href="https://app.graphite.dev/github/pr/DataDog/datadog-agent/14896" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/datadog-agent/15153?utm_source=stack-comment).

modernplumbing committed:
release note

dependabot[bot] committed:
Bump github.com/Microsoft/go-winio from 0.5.2 to 0.6.0 (#13728)

* Bump github.com/Microsoft/go-winio from 0.5.2 to 0.6.0

Bumps [github.com/Microsoft/go-winio](https://github.com/Microsoft/go-winio) from 0.5.2 to 0.6.0.
- [Release notes](https://github.com/Microsoft/go-winio/releases)
- [Commits](https://github.com/Microsoft/go-winio/compare/v0.5.2...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/Microsoft/go-winio
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Auto-generate go.sum and LICENSE-3rdparty.csv changes

* go mod tidy

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>
Co-authored-by: Brian Floersch <brian.floersch@datadoghq.com>

L3n41c committed:
Factorize all the functions creating a pointer (#15118)

EmmaOnThursday committed:
Logging and pipeline changes to address failures to send data to DBM endpoints (#15058)

* epforwarder: double dbm-samples defaultBatchMaxContentSize

We have been seeing some issues where data fails to send
to our dbm samples endpoint. We see far fewer examples of
the dbm-activity and dbm-metrics pipelines being blocked or
failing to send data, and we suspect that this is partly
due to the drastic differences in size between those
pipelines and the dbm-samples pipeline.

This change doubles the defaultBatchMaxContentSize for
the dbm-samples pipeline to try to address this.

* batch_strategy: add debug logging for dbm pipelines

* http/destination: enrich logmessages about sleep backoff

* logs...http/destination: additional logging for err reponses

* logs: new metric to capture err status codes from http resp

brycekahle committed:
Update ebpf-manager (#15174)

lebauce committed:
Send failed report first (#14700)

maycmlee commented on releasenotes/notes/runtime-security-config-file-handling-a9508fda817b9b30.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+enhancements:
+  - |
+    The security agent will fail quickly if the first of either the default, or if they exist, user-provided override config files are inaccessible.
+    The default config files are still datadog.yaml and security-agent.yaml, in that order.
+    Previously, if the security agent was started with at least one accessible config file, no matter the order, the agent would still be able to start.
```
Can we update this to what was decided on here? https://github.com/DataDog/datadog-agent/pull/14814#pullrequestreview-1261831310

modernplumbing replied:
sorry about this, this change was stacked on top of the previous change and then i did something weird in git!

ameske committed:
[RCM] Allow agent clients to specify their k8s cluster (#15148)

This allows predicates to be written and configurations to be targeted
towards agent clients running on a particular cluster.

L3n41c committed:
Container image metadata check: add layers history (#15150)

iglendd committed:
Add to permissions.log windows permissions (#14866)

Add to permissions.log windows permissions
